src/Commands/RebuildCaches.php

```diff
@@ -0,0 +1,112 @@
+<?php
+namespace Eloquence\Commands;
+
+use Eloquence\Behaviours\CountCache\CountCache;
+use Eloquence\Behaviours\SumCache\SumCache;
+use hanneskod\classtools\Iterator\ClassIterator;
+use Illuminate\Console\Command;
+use Illuminate\Database\Eloquent\Model;
+use Symfony\Component\Finder\Finder;
+
+class RebuildCaches extends Command
+{
+
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'eloquence:rebuild {--class= : Optional classes to update}';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Rebuild the caches for one or more eloquent models';
+
+    /**
+     * Create a new command instance.
+     */
+    public function __construct()
+    {
+        parent::__construct();
+    }
+
+    /**
+     * Execute the console command.
+     *
+     * @return mixed
+     */
+    public function handle()
+    {
+
+        if ($class = $this->option('class')) {
+            $classes = [$class];
+        } else {
+            $classes = $this->getAllCacheableClasses();
+        }
+        foreach ($classes as $className) {
+            $this->rebuild($className);
+        }
+    }
+
+    /**
+     * Iterate through the application's classes and return all the ones
+     * that implement one of the caching behaviours.
+     *
+     * @return array[string]
+     */
+    private function getAllCacheableClasses()
+    {
+        $finder = new Finder;
+        $iterator = new ClassIterator($finder->in(app_path()));
+        $iterator->enableAutoloading();
+
+        $classes = [];
+
+        foreach ($iterator->type(Model::class) as $className => $class) {
+            echo "...$className\n";
+            if ($class->isInstantiable() && $this->usesCaching($class)) {
+                $classes[] = $className;
+            }
+        }
+
+        return $classes;
+    }
+
+    /**
+     * Rebuilds the caches for the given class.
+     *
+     * @param string $className
+     */
+    private function rebuild($className)
+    {
+        $instance = new $className;
+
+        if (method_exists($instance, 'countCaches')) {
```

Comment:
Even so, can we extract this out to its own service, maybe using a strategy pattern as well? That way if we implement further cache solutions (which is certainly in the pipeline), we'll be adhering to the O principle in SOLID, and allow for easy extension.